File: findDuplicatesO(n).js
var findDuplicates = function(nums) {
    res = [];
    for(num of nums){
        if(nums[Math.abs(num) - 1] < 0) {
            res.push(Math.abs(num));
        } else {
            nums[Math.abs(num) - 1] *= -1;
        }
    }
    return res;
};

File: findDuplicatesO(n).py
 def findDuplicates(self, nums: List[int]) -> List[int]:
        r = []

        for x in nums:
            if nums[abs(x) - 1] < 0:
                r.append(abs(x))
            else:
                nums[abs(x) - 1] *= -1
        print(nums)
        return (r)